### ISSUE TITLE
Training code available?

### ISSUE THREAD

**Issue by npuichigo** [SandAI-org/MAGI-1#79]

**Comment by npuichigo** [comment]
@stupidZZ thanks for the response. I just want to know do you use video with fixed length in training? Or if they have variable length, do you randomly select the range to inject noise (like 8 seconds), but let the attention to be able to attend to the clean chunks before, and only calculate loss for this part as the paper said? I'm really interested to experiment on speech, but long speech is hard to work well with the non-decending timestep trick, maybe I need also to select several intermediate chunks to do that.

**Comment by stupidZZ** [comment]
Hi @npuichigo, we randomly sample the video length during training and adopt the Magi-attention to support efficient training.

**Comment by npuichigo** [comment]
Thanks. If the length is variable, to support non-descending time steps, the time step granularity of long sequences is smaller. For example, a sequence length > 1000 would have almost all time steps. Will this have any bad effect?
